### DISCUSSION TITLE
Convert IImage back to SKBitmap

### DISCUSSION THREAD

**Discussion by ejongejans** [AvaloniaUI/Avalonia#9045] [Q&A] [answered]
Hi, 

I would like to convert an IImage coming from an Image control back to an SKBitmap in code. 
Reason: our entire backend code base is based on Skia. 
I'm a bit lost in all interfaces of images and contexts. 

Current 2 options i found below; but they don't compile. What is the missing link? 

Anybody any idea? 

```
        public static SKBitmap ToSKBitmap(this IImage img)
        {
            Rect r = new Rect(img.Size);

            SKBitmap bmp = SKBitmap(img.Size.Width,img.Size.Height);
            using (SKCanvas g = new SKCanvas(bmp))
            {
                IDrawingContextImpl contextImpl = DrawingContextHelper.WrapSkiaCanvas(g, new Vector(96, 96));

                // idea 2: 
                contextImpl.DrawBitmap(img, 1, r, r); // ERROR can't convert IImage -> IBitmapImpl  

                // idea 1: 
                img.Draw(contextImpl, r, r, BitmapInterpolationMode.HighQuality); //ERROR can't convert IDrawingContextImpl -> DrawingContext
            }

            return bmp;                
        }
```

**Comment by workgroupengineering** [comment, marked as answer]
Have you tried with [RenderTargetBitmap](http://reference.avaloniaui.net/api/Avalonia.Media.Imaging/RenderTargetBitmap/) ? See [this](https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Diagnostics/Diagnostics/VisualExtensions.cs) for example how to use.

**Reply by ejongejans** [reply]
That only takes an IControl as argument to bitmap.Render ? I don't really care about the control itself; just getting back the data it's displaying (in this case the image) 

How would one then use the IImage ?

**Reply by Gillibald** [reply]
It is not ideal but you should be able to wrap the `IDrawingContextImpl` with `new DrawingContext(contextImpl)` and render the source `IImage` that way

**Reply by ejongejans** [reply]
Great that works! For future reference: 

            using (SKCanvas g = new SKCanvas(bmp))
            {
                IDrawingContextImpl contextImpl = DrawingContextHelper.WrapSkiaCanvas(g, new Vector(96, 96));
                DrawingContext cnt = new DrawingContext(contextImpl);
                iimage.Draw(cnt, r, r, BitmapInterpolationMode.HighQuality);
            }
